### PR TITLE
FI-276 Fetch search parameters more robustly

### PR DESCRIPTION
**Pull request by czh-orz**
This change will make it so each search after the first will look at all the resources in the bundle for a search value. It will also look through more values in searches against datatypes with multiple values (like address or humanname or period). 

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR:https://oncprojectracking.healthit.gov/support/browse/FI-276
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [X] Tests are included and test edge cases
- [X] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/app/sequence_base.rb [hunk 1]
```diff
@@ -725,6 +725,27 @@ def resolve_element_from_path(element, path)
...
+        when FHIR::Reference
+          element.reference
+        when FHIR::CodeableConcept
+          resolve_element_from_path(element, 'coding.code')
+        when FHIR::Identifier
+          element.value
+        when FHIR::Coding
+          element.code
+        when FHIR::HumanName
+          element.family || element.given || element.text
```
Comment (Jammjammjamm) [inline lib/app/sequence_base.rb]: `given` is an array, so I think we'd need `given.first`.

File: test/unit/sequence_base_test.rb [hunk 2]
```diff
@@ -48,4 +48,44 @@ def set_resource_reference(resource, type)
...
+
+  describe '#retrieves value for search param' do
+    before do
+      instance = Inferno::Models::TestingInstance.create(selected_module: 'uscore_v3.0.0')
+      client = FHIR::Client.new('')
+      @sequence = Inferno::Sequence::SequenceBase.new(instance, client, true)
+    end
+
+    it 'returns value from period' do
+      element = FHIR::Period.new
```
Comment (Jammjammjamm) [inline test/unit/sequence_base_test.rb]:
```suggestion
      element = FHIR::Period.new(start: 'now')
```
Would this sort of thing work for all these? If so, you could get a little fancy and really DRY things up by creating a hash of all the options you wanted to test and iterating over it.
```
{ state: mass, ... }.each do |key, value|
   element = FHIR::Address.new(key => value)
   ...
end
```

Comment (czh-orz) [reply test/unit/sequence_base_test.rb]: Did it the DRY way, great suggestion.

File: lib/app/sequence_base.rb (hunk 1 above)
Comment (czh-orz) [reply lib/app/sequence_base.rb]: fixed

File: test/unit/sequence_base_test.rb [hunk 3]
```diff
@@ -48,4 +48,44 @@ def set_resource_reference(resource, type)
     new_reference.reference = "#{type}/1234"
     resource.recorder = new_reference
   end
+
+  describe '#retrieves value for search param' do
```
[the remark this answers is not in the document]
Comment (czh-orz) [reply test/unit/sequence_base_test.rb]: fixed

**Pull request merged**